src/pipecat/processors/user_idle_processor.py

```diff
@@ -42,24 +54,60 @@ async def handle_idle(processor: "UserIdleProcessor") -> None:
     def __init__(
         self,
         *,
-        callback: Callable[["UserIdleProcessor"], Awaitable[None]],
+        callback: Union[
+            Callable[["UserIdleProcessor"], Awaitable[None]],  # Basic
+            Callable[["UserIdleProcessor", int], Awaitable[bool]],  # Retry
+        ],
         timeout: float,
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self._callback = callback
+        self._callback = self._wrap_callback(callback)
         self._timeout = timeout
+        self._retry_count = 0
         self._interrupted = False
         self._conversation_started = False
         self._idle_task = None
         self._idle_event = asyncio.Event()
 
-    def _create_idle_task(self):
-        """Create the idle task if it hasn't been created yet."""
+    def _wrap_callback(
+        self,
+        callback: Union[
+            Callable[["UserIdleProcessor"], Awaitable[None]],
+            Callable[["UserIdleProcessor", int], Awaitable[bool]],
+        ],
+    ) -> Callable[["UserIdleProcessor", int], Awaitable[bool]]:
+        """Wraps callback to support both basic and retry signatures.
+
+        Args:
+            callback: The callback function to wrap.
+
+        Returns:
+            A wrapped callback that returns bool to indicate whether to continue monitoring.
+        """
+
+        async def wrapper(processor: "UserIdleProcessor", retry_count: int) -> bool:
+            if len(callback.__code__.co_varnames) == 1:
```

Comment:
If I am not mistaken, `co_varnames` can include all local variables inside the callBack function, that could lead to mistakes here. 
Maybe refactor to use `co_argcount` that should ignore the local variables. 

Or even better, try to use `inspect.signature`: https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object